### PR TITLE
[14.0][ADD] resource_booking_sale

### PR DESCRIPTION
**Pull request by carmenbianca**
## Odoo task (if applicable)

https://gestion.coopiteasy.be/web#id=9832&action=475&active_id=434&model=project.task&view_type=form&menu_id=536

## Checklist before approval

- [x] Fix the name.
- [x] Make the sale order confirmation button prettier.
- [ ] Should the sale order line with the booking product be protected/special?
- [x] Tests are present (or not needed).
- [ ] Credits/copyright have been changed correctly.
- [ ] Change log snippet is present.
- [ ] (If a new module) Moving this to OCA has been considered.

### CODE REVIEW

**Comment by carmenbianca** [comment]
I need to document this somewhere.

Previously, when creating a new sale order line and selecting a product, the price incorrectly defaulted to 0. See video.

[before.webm](https://user-images.githubusercontent.com/12065945/216073617-6dcab26e-cc67-4933-9ce1-5f756c2530ce.webm)

But after [35f90ab](https://github.com/coopiteasy/addons/pull/277/commits/35f90abdc5b36033dfb89a73abcff77465bb448e), this behaviour was fixed.

Diff:

```diff
diff --git a/sale_resource_booking2/views/resource_booking_views.xml b/sale_resource_booking2/views/resource_booking_views.xml
index 3a86374..1a66ffc 100644
--- a/sale_resource_booking2/views/resource_booking_views.xml
+++ b/sale_resource_booking2/views/resource_booking_views.xml
@@ -62,16 +62,20 @@
                     -->
                     <tree string="Sales Order Lines" editable="bottom">
                         <control>
-                            <create name="add_product_control" string="Add a product" />
+                            <create
+                                name="add_product_control"
+                                string="Add a product"
+                                context="{'default_order_id': parent.sale_order_id}"
+                            />
                             <create
                                 name="add_section_control"
                                 string="Add a section"
-                                context="{'default_display_type': 'line_section'}"
+                                context="{'default_display_type': 'line_section', 'default_order_id': parent.sale_order_id}"
                             />
                             <create
                                 name="add_note_control"
                                 string="Add a note"
-                                context="{'default_display_type': 'line_note'}"
+                                context="{'default_display_type': 'line_note', 'default_order_id': parent.sale_order_id}"
                             />
                         </control>
```

Video:

[after.webm](https://user-images.githubusercontent.com/12065945/216073883-176adf69-6f8f-4cff-81cb-ad4761ef0f04.webm)

This change in behaviour makes no sense to me.

**Comment by carmenbianca** [comment]
Retroactively renamed the module to resource_booking_sale. Rewriting history is _powerful_.

File: resource_booking_sale/models/resource_booking.py [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+                lambda line: line.product_id == booking.product_id
+            )
+            if booking.product_id and not booking_line:
+                self.env["sale.order.line"].create(
+                    {
+                        # TODO: Verify this.
+                        "name": _("Booking for %s") % booking.partner_id.name,
+                        "sequence": 1,
+                        "product_id": booking.product_id.id,
+                        "order_id": booking.sale_order_id.id,
```
Comment (remytms) [inline resource_booking_sale/models/resource_booking.py]: Has it been verified ?

File: resource_booking_sale/models/sale_order.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+    resource_booking_ids = fields.One2many(
+        comodel_name="resource.booking",
+        inverse_name="sale_order_id",
+        string="Resource Booking",
+    )
+
+    def action_cancel(self):
+        # TODO: cancel resource_booking_ids. Verify before implementing.
+        return super().action_cancel()
```
Comment (remytms) [inline resource_booking_sale/models/sale_order.py]: Has it been verified ?

File: resource_booking_sale/readme/DESCRIPTION.rst [hunk 3]
```diff
@@ -0,0 +1 @@
+An alternative implementation of sale_resource_booking (v12).
```
Comment (remytms) [inline resource_booking_sale/readme/DESCRIPTION.rst:5]: This information is nice. But explain nothing on what the module does. At least a link to sale_resource_booking is welcome. :)

**Review by remytms — APPROVED** [review]
LGTM. But some questions on some `TODO` in the code.

File: resource_booking_sale/models/resource_booking.py (hunk 1 above)
Comment (carmenbianca) [reply resource_booking_sale/models/resource_booking.py]: this yes

File: resource_booking_sale/readme/DESCRIPTION.rst (hunk 3 above)
Comment (carmenbianca) [reply resource_booking_sale/readme/DESCRIPTION.rst:5]: fixed

**Pull request merged**